src/main/recorder.ts

```diff
@@ -51,6 +59,48 @@ type RecorderOptionsType = {
         if (mainWindow) mainWindow.webContents.send('refreshState');
     }
 
+    /**
+     * Process an item from the video queue
+     */
+    private async _processVideoQueueItem(data: VideoQueueItem, done: Function): Promise<void> {
+        const outputFilename = this.getFinalVideoFilename(data.metadata);
+        const videoPath = await cutVideo(data.bufferFile, this._options.storageDir, outputFilename, data.metadata.duration);
+        await writeMetadataFile(videoPath, data.metadata);
+
+        // Delete the original buffer video
+        tryUnlinkSync(data.bufferFile);
+
+        done();
+    }
+
+    /**
+     * Setup events on the videoQueue
+     */
+    private _setupVideoProcessingQueue(): void {
+        this._videoQueue.on('error', (err: any) => {
+            console.error("[Recorder] Error occured during video processing", err);
+        });
+
+        this._videoQueue.on('idle', () => {
+            console.log("[Recorder] Video processing queue empty, running clean up.")
+
+            // Run the size monitor to ensure we stay within size limit.
+            // Need some maths to convert GB to bytes
+            runSizeMonitor(this._options.storageDir, this._options.maxStorage)
```

Comment:
Should this be in the finish event?

Reply:
I find it more suitable to run it once the queue is idle/empty, but I honestly don't care when it's run. :)

Reply:
Only slight preference as this way we avoid some condition where the queue gets backed up and we temporarily exceed the storage limit. Not going to lose sleep over it either way.

Reply:
I don't care enough about this in retrospect to hold up the PR. Lets get this in.